trl/trainer/gkd_config.py

```diff
@@ -33,6 +33,8 @@ class GKDConfig(SFTConfig):
             beta is `0.0`, the loss is the KL divergence. When beta is `1.0`, the loss is the Inverse KL Divergence.
         max_new_tokens (`int`, *optional*, defaults to `128`):
             Maximum number of tokens to generate per completion.
+        eos_token_id (`Union[int, List[int]]`, *optional*):
```

Comment:
In `transformers` the supported types are `int` and `List[int]` but the HF arg parser only accepts `Union[X, NoneType]`. Should I refactor the arg parser or just abuse the types a bit here?

Reply:
Hmm I have never tried, `Optional[Union[int, List[int]]]`  is not possible? Otherwise just use Optional[List[int]]? and for the case with 1 token use a list of size 1?

Also unless am referring to the wrong model, these tokens are exposed in the [llama3 gen config](https://huggingface.co/meta-llama/Llama-3.1-8B-Instruct/blob/main/generation_config.json) ?

Reply:
You can do the as https://github.com/huggingface/trl/pull/2108

Reply:
```suggestion
        eos_token_id (`Union[int, List[int]]`, *optional*, default to `None`):
```

Reply:
Ah nice, thanks for the pointer @qgallouedec